app/models/post.rb

```diff
@@ -1,8 +1,14 @@
 class Post < ApplicationRecord
-validates :user, presence: true
-validates :body, presence: true, length: { maximum: 200 }
-
   belongs_to :user
-  has_many :comments
-  has_many :likes
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  validates :body, presence: true, length: { maximum: 200 }
+
+  scope :with_comments_and_likes, -> { includes(:user, :comments, :likes) }
+  scope :user_friends, lambda { |user|
```

Comment:
Please tell me what this scope will be doing later on in the app? Just by looking at it I can see it's going to make a bit of a complex code in the controller later on.